pyinteraph/centrality_analysis.py

```diff
@@ -0,0 +1,604 @@
+                # Find index
+                identifiers.index(res)
+                # Append to list
+                res_list.append(res)
+            except:
+                raise ValueError(f"Residue not in PDB: {res}")
+        # If list size is 2, word is separated by a single colon
+        elif len(sub_list) == 2:
+            try:
+                # Get indexes for residue in identifiers
+                u_idx, v_idx = [identifiers.index(res) for res in sub_list]
+            except:
+                raise ValueError(f"Residue range not in PDB: {':'.join(sub_list)}")
+            # Check if order of residues is reversed
+            if u_idx >= v_idx:
+                raise ValueError(f"Range not specified correctly: {':'.join(sub_list)}")
+            try:
+                # This block should not cause an error
+                # Create list with all residues in that range
+                res_range = identifiers[u_idx:v_idx + 1]
+                # Concatenate with output list
+                res_list += res_range
+            except Exception as e:
+                raise e
+        # Other list sizes means multiple colons
+        else:
+            err_str = f"Incorrect format, only one ':' allowed: {':'.join(sub_list)}"
+            raise ValueError(err_str)
+    # Remove duplicates
+    res_list = list(set(res_list))
+    return res_list
+
+def get_hubs(G, **kwargs):
+    """Returns a dictionary of degree values for each node."""
+
+    degree_tuple = G.degree()
+    hubs = {n : (d if d >= kwargs['hub'] else 0) for n, d in degree_tuple}
+    return hubs
+
+def get_degree_cent(G, **kwargs):
+    """Returns a dictionary of degree centrality values for each node."""
+
+    centrality_dict = nxc.degree_centrality(G)
+    return centrality_dict
+
+def get_betweeness_cent(G, **kwargs):
+    """Returns a dictionary of betweeness centrality values for each node."""
+
+    # Need to consider if endpoints should be used or not
+    centrality_dict = nxc.betweenness_centrality(G = G,
+                                                 normalized = kwargs['norm'],
+                                                 weight = kwargs['weight_name'],
+                                                 endpoints = kwargs['endpoint'])
+    return centrality_dict
+
+def get_closeness_cent(G, **kwargs):
+    """Returns a dictionary of closeness centrality values for each node."""
+
+    centrality_dict = nxc.closeness_centrality(G = G)
+    return centrality_dict
+
+
+# def get_communicability_betweenness_cent(G, **kwargs):
+#     """Returns a dictionary of communicability betweenness centrality values"""
+
+#     centrality_dict = nxc.communicability_betweenness_centrality(G = G,
+#                                                                  normalized = kwargs['norm'])
+#     return centrality_dict
+
+def get_dict_with_group_val(G, node_list, value):
+    """
+    Takes in a graph, list of nodes and a group value. Returns a dict
```

Comment:
Move the beginning of the description in the same line where the leading `"""` is, as done for the other functions' descriptions.